### ISSUE TITLE
Generated results do not match those in the paper / website.

### ISSUE THREAD

**Issue by mitchellnw** [facebookresearch/low-shot-shrink-hallucinate#6]
Hello, when running your code I am not able to reproduce your results.

For example, using ResNet10 with no auxiliary loss and no generation, these are my results.

A difference of 8 points for n=1 on Novel Top-1 seems very large. Also, this baseline model performs a lot worse than reported large n.

n 	Novel 	Novel 	Base 	Base 	All 	All
  	Top-1 	Top-5 	Top-1 	Top-5 	Top-1 	Top-5
=====================================================
1	10.05	35.00	72.58	88.29	34.22	55.60
2	17.12	48.77	70.82	86.70	37.88	63.43
5	23.97	59.94	70.30	86.18	41.88	70.09
10	26.69	64.39	70.58	86.12	43.65	72.79
20	28.37	66.75	70.84	86.19	44.79	74.27
=====================================================
mean	21.24	54.97	71.02	86.70	40.48	67.24

For reproducibility, I used the following command (and pytorch 0.4.0):

```
python ./main.py --model ResNet10 \
  --traincfg base_classes_train_template.yaml \
  --valcfg base_classes_val_template.yaml \
  --print_freq 10 --save_freq 10 \
  --checkpoint_dir checkpoints/ResNet10_sgm
```

**Comment by bharath272** [comment]
Can you tell me what results you get if you use the trained model we've
uploaded?

**Comment by mitchellnw** [comment]
Hi Bharath,

Sure -- I will post them within the hour.

**Comment by mitchellnw** [comment]
Sorry for the delay I am having some issues with my computer. Will post the results as soon as I can.

**Comment by mitchellnw** [comment]
Sorry for the delay, here are the results using your pertained model.

```
n 	Novel 	Novel 	Base 	Base 	All 	All
  	Top-1 	Top-5 	Top-1 	Top-5 	Top-1 	Top-5
=====================================================
1	1.02	8.95	75.80	91.75	29.93	40.96
2	5.10	24.00	75.62	91.33	32.36	50.03
5	18.34	48.05	74.21	90.08	39.94	64.30
10	29.92	61.24	72.09	88.68	46.22	71.85
20	39.62	69.90	69.48	87.04	51.17	76.53
=====================================================
mean	18.80	42.43	73.44	89.78	39.92	60.73
```

**Comment by mitchellnw** [comment]
As an additional note on the results above, these match exactly with the LogReg findings from your 2018 paper 'Low-Shot Learning from Imaginary Data'. However, they fall short of the Baseline Classifier from the Low Shot Shrink Hallucinate paper. I was wondering if you would be able to clarify this if possible.

Thank you so much for your time and help.

**Comment by bharath272** [comment]
Ok, so the difference that you are observing with the provided model is because of hyperparameter choices during low-shot learning, in particular the weight-decay. To reproduce the numbers in the paper, you'll need to use a weight decay of 0.01 instead of the default of 0.001. You can do this by adding "--wd 0.01". This was obtained through cross-validation (I'll need to check why our two papers seem to have found different values for this hyperparameter; will get back to you on that). Unfortunately, this hyperparameter ends up quite significantly affecting the balance between the base and novel classes. 

The numbers you report after training your own model are significantly better than the numbers we got  in terms of all-class accuracy. This seems strange. I'll rerun the code on my end and see what I get.

Thanks for bringing this to my attention!

**Comment by mitchellnw** [comment]
Thank you! I will rerun everything to verify and try using the new weight decay value.

**Comment by bharath272** [comment]
(Just checking: in the set of commands that you wrote in your initial post, you've used "checkpoints/ResNet10_sgm" during model training, but "checkpoints/ResNet10" when saving the features. I am assuming this is just a typo when copying here?)

**Comment by mitchellnw** [comment]
Yes sorry that is just a typo!

**Comment by mitchellnw** [comment]
Thank you, with the model you've provided and new hyperparameters I was able to reproduce the values from the Low Shot Shrink Hallucinate paper. 

However, these results still differ from those in the Imaginary Data paper. Namely, for n=1, top 5 accuracy on novel classes, The Imaginary Data paper reports a 38.4 accuracy while the Low Shot Shrink Hallucinate paper reports 14.1. Perhaps I am misunderstanding something here and the experiment has changed -- if so please let me know.

**Comment by bharath272** [comment]
The benchmark is different. If you look at the imaginary data paper, we
point out that because hyperparameters can have a big impact on the balance
between base and novel classes, we report "novel class performance" in
isolation, instead of in conjunction with base classes. This is explained
in Section 5, under the subheading "A new evaluation".

**Comment by mitchellnw** [comment]
Hi Bharath,

Thank you very much for all your help -- all of my issues are resolved.
